### PR TITLE
Added hermit commands to deal with profile git remote addresses

### PR DESCRIPTION
**Pull request by aig787**

### CODE REVIEW

File: lib/hermit-help [hunk 1]
```diff
@@ -101,8 +101,15 @@ print_summary() {
   local summary usage help
   eval "$(documentation_for "$command")"
 
+  # Max number of characters we can print in summary without passing 80 total
+  summary_max=$(expr 80 - 9 - ${#command})
+
   if [ -n "$summary" ]; then
-    printf "   %-9s   %s\n" "$command" "$summary"
+    if [ "${#summary}" -gt "$summary_max" ]; then
+      printf "   %-20s   %s...\n" "hermit $command" "${summary:0:$summary_max}"
```
Comment (RadicalZephyr) [inline lib/hermit-help]: I like the idea of wrapping the summaries, but I'd rather wrap them than truncate.  Maybe remove this change and we'll address it in a separate PR.

File: lib/hermit-remote-remove [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+echo
+
+pushd $HERMIT_PROFILE_PATH >/dev/null 2>/dev/null
+
+if [ -n "$1" ]; then
+  echo "Removing remote $1"
+  git remote remove $1
+else
+  echo "Remote remote origin"
+  git remtoe remove origin
```
Comment (RadicalZephyr) [inline lib/hermit-remote-remove]: Don't think you meant "Remote remote"

And remtoe -> remote

**Comment by aig787** [comment]
@RadicalZephyr should be fixed now

File: lib/hermit-remote-remove [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+echo "Located at: $HERMIT_PROFILE_PATH"
+echo
+
+pushd $HERMIT_PROFILE_PATH >/dev/null 2>/dev/null
+
+if [ -n "$1" ]; then
+  echo "Removing remote $1"
+  git remote remove $1
+else
+  echo "Remote remove origin"
```
Comment (RadicalZephyr) [inline lib/hermit-remote-remove]: Shouldn't it be "Remove remote origin"?

File: lib/hermit-remote-remove [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  git remtoe remove origin
```
Comment (RadicalZephyr) [inline lib/hermit-remote-remove]: Still have remtoe :)

**Comment by reillysiemens** [comment]
So, `hermit remote` is a new hermit subcommand? Does it have tests to go with it, @aig787?

**Comment by RadicalZephyr** [comment]
Looks much better :+1: Now squash into one commit, and @reillysiemens is right, please add some basic sanity tests.

I'm just thinking though, is it worth it to wrap remote at that granularity?  I'm wondering if this use case might be better served by bringing back the idea of a `hermit git` command where you can give arbitrary `git` commands.  I had some notes about why I thought that was a bad idea at the time I implemented it, but apparently I deleted the branch that contained them...

File: lib/hermit-remote-add [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with Hermit. If not, see <http://www.gnu.org/licenses/>.
+
+set -e
+[ -n "$HERMIT_DEBUG" ] && set -x
+
```
Comment (RadicalZephyr) [inline lib/hermit-remote-add]: We also need to do a check for if this is being called for completions.  You don't need to write the code that will give the proper completions, but you do need to exit early if that's the case.  Check out `hermit use` for what that code looks like.

**Comment by reillysiemens** [comment]
Personally I wouldn't want hermit to be concerned with git at this granularity. I'd rather implement a `hermit show` that does a `pushd` into your `$HERMIT_DIR` and then you can do whatever you want with git directly and pop right back out.

**Comment by RadicalZephyr** [comment]
We talked about this before in #18.  The problem with it is that in order to implement changing the cwd all of hermit needs to be implemented as shell functions and sourced or eval'd into the user's shell.

@aig787 what do you think about having a `hermit open` or `hermit show` command like @reillysiemens described?

**Comment by aig787** [comment]
Maybe, but working with remotes seems like a core piece of the functionality here. If we aren't providing a way to deal with taking your home directory with you the whole project loses quite a bit of appeal.

**Comment by RadicalZephyr** [comment]
That's a fair point.  Since I already had my dotfiles in git, the main use case I've been concerned with was cloning down an existing repo.  But if you're using `hermit init` to start a dotfiles repo, then you do need to be able to set a remote.  But arguably then we'd need a `hermit push`.

I think the distinction that I see is that `hermit` is a facilitator.  The ability to manage your dotfiles across different machines really comes directly from git. And git is such a powerful tool that I personally wouldn't want to try and wrap it.  Especially since the wrapping is then limited.

How about this.  Instead of wrapping the remote subcommands directly, let's provide a generic remote wrapper.  It would still be invoked like `hermit remote add origin <git-place>` but the difference is that everything after `hermit` would just be passed along wholesale to git.  Thus the script would *only* be a way to pass through arbitrary (and well-formatted) git commands.

**Comment by aig787** [comment]
That would be a good way to do it I think. I thought about doing it that way myself but decided for what I would use it for more granularity made more sense.

**Comment by aig787** [comment]
I'm going to close this and re-open with the pass through approach we talked about.

**Comment by RadicalZephyr** [comment]
Sounds good.  Looking forward to reviewing it.

**Pull request closed without merging**